### PR TITLE
Dragging behavior should not jump around.

### PR DESCRIPTION
**Pull request by jcoyne**
Also converted Selection to top, bottom, left, right instead of x,y,height,width as it simplifies the code dramatically.

### CODE REVIEW

File: src/move.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+'use strict';
```
Comment (aeschylus) [inline src/move.js:1]: `move` and `resize` do not need to be objects. They should be functions on the selection object that update the state there.

File: src/move.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+var Move = function(state) {
+  this.state = state;
+  this.lastPosition = {};
```
Comment (aeschylus) [inline src/move.js]: When moving this to the selection object, you can have a local variable by using `var myVariable` inside the function. Javascript has function scope (a function declaration creates a closure), so as long as you use the `var` keyword you can access to a `lastPosition` variable within that closure without polluting the scope of the selection object if you need it.

``` javascript

// "a" will be in the global namespace
a = "a";

// "a" will be inside this module/object/whatever contains this example script (as a whole). "a" will overwrite the outer "a" and will have the value "b". Variables using `var` shadow each outer declarations and so on down into function and object nestings.
var a = "b";

myFunction = function() {
var a = "c";
console.log(a);
}

console.log(a); // will return "b" from above. The inner "a" does not overwrite this.

```

File: src/selection.js [hunk 3]
```diff
@@ -1,10 +1,10 @@
 'use strict';
 
 var Selection = function(options, dispatcher) {
-  this.x = 0 || options.x;
-  this.y = 0 || options.y;
-  this.width = 0 || options.width;
-  this.height = 0 || options.height;
+  this.top = 0 || options.top;
```
Comment (aeschylus) [inline src/selection.js:4]: 👍

File: src/move.js (hunk 1 above)
Comment (jcoyne) [inline src/move.js:1]: Yes, they do not need to be objects. But is there any reason they can't be objects?  I find that having them as objects makes the code more testable.

Comment (jcoyne) [inline src/move.js:1]: I would classify the Selection object as a Model object.  The Move and Resize are ViewController objects. The shouldn't be squashed together.

**Pull request merged**